equinix/resource_metal_spot_market_request.go

```diff
@@ -22,6 +23,10 @@ import (
 	"github.com/packethost/packngo"
 )
 
+var (
+	matchIPXEScript = regexp.MustCompile(`(?i)^#![i]?pxe`)
```

Comment:
Chose to duplicate this matcher instead of finding a place to put it.  IMO we can clean that up after framework and equinix-sdk-go migrations are done.

Reply:
I don't think the validation this enforces is necessary in the Terraform provider. Perhaps https://github.com/equinix/terraform-provider-equinix/blob/21c9dcc8dcd6eff9075b8c29cdb5fa57f170befa/equinix/resource_metal_spot_market_request.go#L274-L277 would be better as a warning or left to the API to enforce.  It's well enough to leave this validation alone since there are no known edge cases where the combination of an IPXE URL and a Custom IPXE Userdata are necessary.

Reply:
Perhaps the regexp could be moved closer to where it is used since it is not used elsewhere.